activestorage/test/jobs/create_variants_job_test.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+require "test_helper"
+require "database/setup"
+
+class ActiveStorage::CreateVariantsJobTest < ActiveJob::TestCase
+  setup do
+    @transformations = [{ resize_to_limit: [ 100, 100 ] }, { resize_to_limit: [ 200, 200 ] }]
+  end
+
+  test "creates preview when previewable" do
+    blob = create_file_blob(filename: "report.pdf", content_type: "application/pdf")
+
+    assert_changes -> { blob.preview_image.attached? }, from: false, to: true do
+      ActiveStorage::CreateVariantsJob.perform_now blob, generate: :delayed, transformations: @transformations
+    end
+  end
+
+  test "enqueues individual transform jobs for each transformation when :delayed" do
+    blob = create_file_blob
+    ActiveStorage::CreateVariantsJob.perform_now blob, generate: :delayed, transformations: @transformations
+
+    @transformations.each do |transformation|
+      assert_enqueued_with job: ActiveStorage::TransformJob, args: [ blob, transformation ]
+    end
+  end
+
+  test "does not transform when :delayed" do
+    blob = create_file_blob
+    ActiveStorage::CreateVariantsJob.perform_now blob, generate: :delayed, transformations: @transformations
+
+    @transformations.each do |transformation|
+      assert_not blob.variant(transformation).send(:processed?)
```

Comment:
I didn't include in this PR, but I think we should consider moving `processed?` to a public method. Not only is it used in testing, but our app uses it as a conditional to determine whether to display a variant on highly requested pages.

Reply:
It seems really weird there's no public way to ask if a variant is processed. I know in Build With Becky I wound up doing something like rescuing an unprocessed error instead

Reply:
Do you think the processed? Method should be public?